resources/sqlite/src/conn_opts.rs

```diff
@@ -0,0 +1,270 @@
+    pub fn busy_timeout(mut self, timeout: Duration) -> Self {
+        self.busy_timeout = timeout;
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.synchronous).
+    pub fn synchronous(self, synchronous: SQLiteSynchronous) -> Self {
+        self.pragma("synchronous", synchronous.as_str())
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.auto_vacuum).
+    pub fn auto_vacuum(self, auto_vacuum: SQLiteAutoVacuum) -> Self {
+        self.pragma("auto_vacuum", auto_vacuum.as_str())
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.page_size).
+    pub fn page_size(self, page_size: u32) -> Self {
+        self.pragma("page_size", page_size.to_string())
+    }
+
+    pub(crate) fn pragma<K, V>(mut self, key: K, value: V) -> Self
+    where
+        K: Into<Cow<'static, str>>,
+        V: Into<Cow<'static, str>>,
+    {
+        self.pragmas.insert(key.into(), Some(value.into()));
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.immutable).
+    pub fn immutable(mut self, immutable: bool) -> Self {
+        self.immutable = immutable;
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.serialized).
+    pub fn serialized(mut self, serialized: bool) -> Self {
+        self.serialized = serialized;
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.command_buffer_size).
+    pub fn command_buffer_size(mut self, size: usize) -> Self {
+        self.command_channel_size = size;
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.row_buffer_size).
+    pub fn row_buffer_size(mut self, size: usize) -> Self {
+        self.row_channel_size = size;
+        self
+    }
+
+    /// See [sqlx docs](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.vfs).
+    pub fn vfs(mut self, vfs_name: impl Into<Cow<'static, str>>) -> Self {
+        self.vfs = Some(vfs_name.into());
+        self
+    }
+}
+
+impl TryFrom<&SQLiteConnOpts> for SqliteConnectOptions {
+    type Error = shuttle_service::Error;
+
+    fn try_from(opts: &SQLiteConnOpts) -> Result<Self, Self::Error> {
+        let SQLiteConnOpts {
+            storage_path,
+            filename,
+            in_memory,
+            read_only,
+            create_if_missing,
+            statement_cache_capacity,
+            busy_timeout,
+            immutable,
+            vfs,
+            serialized,
+            command_channel_size,
+            row_channel_size,
+            ..
```

Comment:
You are right. Now I remember that at some piont I realised that I don't have an easy way to pass pragmas to the final config we're creating in `try_from`. I should have removed the whole pragma business I guess.

This is actually a good argument for exposing only a limited set of options. For now, we could simply expose all the options that are being set directly and remove the pragma logic completely. This would actually not change what is being exposed right now as I already put a disclaimer in the docs that pragmas are not exposed.